com.unity.render-pipelines.universal/Runtime/Passes/GBufferPass.cs

```diff
@@ -24,12 +26,31 @@ public GBufferPass(RenderPassEvent evt, RenderQueueRange renderQueueRange, Layer
             m_FilteringSettings = new FilteringSettings(renderQueueRange, layerMask);
             m_RenderStateBlock = new RenderStateBlock(RenderStateMask.Nothing);
 
+            m_RenderStateBlock.stencilReference = stencilReference;
+            m_RenderStateBlock.mask = RenderStateMask.Stencil;
             if (stencilState.enabled)
             {
-                m_RenderStateBlock.stencilReference = stencilReference;
-                m_RenderStateBlock.mask = RenderStateMask.Stencil;
                 m_RenderStateBlock.stencilState = stencilState;
             }
+            else
+            {
+                m_RenderStateBlock.stencilState = new StencilState(
+                    true,
+                    0, 0,
+                    CompareFunction.Always, StencilOp.Replace, StencilOp.Keep, StencilOp.Keep,
+                    CompareFunction.Always, StencilOp.Replace, StencilOp.Keep, StencilOp.Keep
+                );
+            }
+
+            m_ShaderTagValues = new ShaderTagId[3];
+            m_ShaderTagValues[0] = new ShaderTagId("Lit");
+            m_ShaderTagValues[1] = new ShaderTagId("SimpleLit");
+            m_ShaderTagValues[2] = new ShaderTagId("Unlit");
+
+            m_RenderStateBlocks = new RenderStateBlock[3];
+            m_RenderStateBlocks[0] = OverwriteStencil(m_RenderStateBlock, 96, 32);
```

Comment:
I really suggest that you create a StencilUsage class for stencilm bit, it is quickly unreadable and you can make a lot of mistake: https://github.com/Unity-Technologies/Graphics/blob/0161c9151e11f854f472cbf7c6903bbb7c90083c/com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDStencilUsage.cs

Also users need an API to access stencil bits from C# side when they do custom pass in C#.

Reply:
Sure, I can replace these hardcoded constants.

Reply:
Refactored into our own StencilUsage enum.